docs/contributing/contributing-checklist.md

```diff
@@ -0,0 +1,121 @@
+Contribution checklist
+Following these instructions helps us make sure the code review and merge process go smoothly.
```

Comment:
Could you have one more run-through of the formatting and links in this document? I'm seeing a number of broken links and formatting that does not match our RTD site

Reply:
Moved in later work to this brnahc